Ginger/GingerCoreCommon/Repository/BusinessFlowLib/BusinessFlow.cs

```diff
@@ -345,6 +345,20 @@ private void DoActivitiesLazyLoad()
                 {
                     this.TrackObservableList(mActivities);
                 }
+
+                IEnumerable<string> distinctTargetApp =  mActivities.Select((activity) => activity.TargetApplication).Distinct();
+
```

Comment:
If now the Activities selected Target Applications set the Business Flow list of Target Applications so no need to maintain “TargetApplications” list on BF level which will be serialized, we can dynamically each time return the current selected Applications on all child Activities.

Reply:
seems like there are many calls to TargetApplicaitons field and also the Activities having only mapping by name so relaying only on Activities can be a problem.